Sources/SwiftLexicalLookup/LookupName.swift

```diff
@@ -121,9 +173,15 @@ import SwiftSyntax
   }
 
   /// Extracts name introduced by `IdentifiableSyntax` node.
-  private static func handle(identifiable: IdentifiableSyntax, accessibleAfter: AbsolutePosition? = nil) -> [LookupName]
-  {
-    if identifiable.identifier.tokenKind != .wildcard {
+  private static func handle(
+    identifiable: IdentifiableSyntax,
+    accessibleAfter: AbsolutePosition? = nil
+  ) -> [LookupName] {
+    if let closureCapture = identifiable as? ClosureCaptureSyntax,
+      closureCapture.identifier.tokenKind == .keyword(.self)
+    {
+      return [.implicit(.self(closureCapture))]  // Handle `self` closure capture.
```

Comment:
You're right, it doesn't really makes sense as the `self` here is explicit. I modeled this here because we didn't have a way to represent non-identifier kind tokens as `Identifier`s. Now, since we can use `String` to create an `Identifier`, we can finally properly [represent this behavior](https://github.com/swiftlang/swift-syntax/blob/252d3be80c010add2bac9f9712ac32d9892ea574/Sources/SwiftLexicalLookup/LookupName.swift#L119).

Reply:
Sounds good!